tests/test-update-install-flatpaks.c

```diff
@@ -2259,6 +2259,260 @@ test_update_flatpaks_no_op_if_not_installed (EosUpdaterFixture *fixture,
+                                          deployment_repo_relative_path);
+  eos_test_setup_flatpak_repo_with_preinstalled_apps (updater_directory,
+                                                      "test-repo",
+                                                      "com.endlessm.TestInstallFlatpaksCollection",
+                                                      (const gchar **) wanted_flatpaks,
+                                                      (const gchar **) wanted_flatpaks,
+                                                      &error);
+  flatpak_build_dir = eos_test_get_flatpak_build_dir_for_updater_dir (updater_directory);
+  flatpak_repo_dir = g_file_get_child (flatpak_build_dir, "repo");
+  flatpak_apps_dir = g_file_get_child (flatpak_build_dir, "apps");
+  app_dir = g_file_get_child (flatpak_apps_dir, flatpaks_to_install[0].app_id);
+  app_dir_path = g_file_get_path (app_dir);
+  flatpak_repo_path = g_file_get_path (flatpak_repo_dir);
+  app_executable_path = g_build_filename (app_dir_path,
+                                          app_executable_relative_path,
+                                          NULL);
+
+  /* Set the contents to something that we expect */
+  g_file_set_contents (app_executable_path, expected_app_executable_contents, -1, &error);
+  g_assert_no_error (error);
+
+  flatpak_build_export (updater_directory, app_dir_path, flatpak_repo_path, &error);
+  g_assert_no_error (error);
+
+  /* Update the server, so it has a new commit (1).
+   */
+  etc_update_server (data, 1);
+  /* Update the client, so it also has a new commit (1); and, at this
+   * point, two deployments - old one pointing to commit 0 and a new
+   * one pointing to commit 1.
+   */
+  etc_update_client (data);
+
+  /* Now simulate a reboot by running eos-updater-flatpak-installer */
+  deployment_csum = get_checksum_for_deploy_repo_dir (deployment_repo_dir,
+                                                      refspec,
+                                                      &error);
+  g_assert_no_error (error);
+
+  eos_test_run_flatpak_installer (data->client->root,
+                                  deployment_csum,
+                                  default_remote_name,
+                                  &error);
+  g_assert_no_error (error);
+
+  /* Assert that our flatpak was updated with new contents */
+  installed_app_executable_path = g_build_filename (flatpak_user_installation,
+                                                    test_flatpak_relative_path,
+                                                    app_executable_relative_path,
+                                                    NULL);
+  g_file_get_contents (installed_app_executable_path,
+                       &installed_app_executable_contents,
+                       NULL,
+                       &error);
+  g_assert_no_error (error);
+
+  g_assert (g_str_equal (installed_app_executable_contents, expected_app_executable_contents));
```

Comment:
Nitpick: This could be `g_assert_cmpstr()` to get a nice error message with both strings in if it fails.

(I’ll fix this before pushing.)